transactions/multi-document-acid-transactions-mongodb-go.md

```diff
@@ -0,0 +1,214 @@
+type Episode struct {
+	ID          primitive.ObjectID `bson:"_id,omitempty"`
+	Podcast     primitive.ObjectID `bson:"podcast,omitempty"`
+	Title       string             `bson:"title,omitempty"`
+	Description string             `bson:"description,omitempty"`
+	Duration    int32              `bson:"duration,omitempty"`
+}
+
+func main() {
+	client, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(os.Getenv("ATLAS_URI")))
+	if err != nil {
+		panic(err)
+	}
+	defer client.Disconnect(context.TODO())
+
+	database := client.Database("quickstart")
+	episodesCollection := database.Collection("episodes")
+}
+```
+
+Now let's also assume that you've correctly included the MongoDB Go driver as seen in a previous tutorial titled, [How to Get Connected to Your MongoDB Cluster with Go](https://www.mongodb.com/blog/post/quick-start-golang--mongodb--starting-and-setup).
+
+The goal here will be to try to insert a document that complies with our schema validation as well as a document that doesn't, so that we have a commit that doesn't happen.
+
+```go
+// ...
+
+func main() {
+    // ...
+
+	session, err := client.StartSession()
+	if err != nil {
+		panic(err)
+	}
+	defer session.EndSession(context.Background())
+
+	err = mongo.WithSession(context.Background(), session, func(sessionContext mongo.SessionContext) error {
+		if err = session.StartTransaction(); err != nil {
+			panic(err)
+		}
+		result, err := episodesCollection.InsertOne(
+			sessionContext,
+			Episode{
+				Title:    "A Transaction Episode for the Ages",
+				Duration: 15,
+			},
+        )
+        fmt.Println(result.InsertedID)
+		result, err = episodesCollection.InsertOne(
+			sessionContext,
+			Episode{
+				Title:    "Transactions for All",
+				Duration: 1,
+			},
+		)
+		if err = session.CommitTransaction(sessionContext); err != nil {
+			panic(err)
+        }
+        fmt.Println(result.InsertedID)
+		return nil
+	})
+	if err != nil {
+		panic(err)
+	}
+}
+```
+
+In the above code we start by starting a session which will encapsulate everything we want to do with atomicity. After, we start a transaction which we'll use to commit everything in the session.
+
+A `Session` represents a MongoDB logical session and can be used to enable casual consistency for a group of operations or to execute operations in an ACID transaction. More information on how they work in Go can be found in the [documentation](https://godoc.org/go.mongodb.org/mongo-driver/mongo#Session).
+
+Inside the session we are doing two `InsertOne` operations. The first would succeed because it doesn't violate any of our schema validation rules. It will even print out an object id when it's done. However, the second operation will fail because it is less than two minutes. The `CommitTransaction` won't ever run because of the error that the second operation created. For this reason, neither of the `InsertOne` operations will show up in the database.
```

Comment:
After adding the `AbortTransaction` logic, modify this paragraph to talk about when/why the transaction is aborted.